domainmessage/msgreject.go

```diff
@@ -0,0 +1,22 @@
+package domainmessage
+
+// MsgReject implements the Message interface and represents a kaspa
+// Reject message. It is used to notify peers why they are banned.
+type MsgReject struct {
+	baseMessage
+	Reason string
```

Comment:
Please make sure the reject message contains the type (Block/Transaction) and Hash/ID of what failed.

I'd rather have it as two separate fields in the message, but if it's already in the Reason string, I don't mind keeping it this way for now.